crates/telio-utils/src/repeated_actions.rs

```diff
@@ -171,6 +166,111 @@ mod tests {
             self.test = str;
             Ok(())
         }
+
+        pub fn get(&self) -> &str {
+            &self.test
+        }
+    }
+
+    #[tokio::test(start_paused = true)]
+    async fn test_set_all_immediate() {
+        let mut ctx = Context::new("test".to_owned());
+
+        let start = Instant::now();
+
+        ctx.actions
+            .add_action(
+                "action_0".to_owned(),
+                Duration::from_secs(10),
+                Arc::new({
+                    let start = start.clone();
+                    move |s: _| {
+                        Box::pin({
+                            let start = start.clone();
+                            async move {
+                                s.change(format!("ts_{}", start.elapsed().as_secs()).to_owned())
+                                    .await
+                            }
+                        })
+                    }
+                }),
+            )
+            .unwrap();
+
+        // immediate action
+        ctx.actions.select_action().await.unwrap().1(&mut ctx)
+            .await
+            .unwrap();
+
+        tokio::time::advance(Duration::from_secs(3)).await;
+
+        ctx.actions
+            .add_action(
+                "action_1".to_owned(),
+                Duration::from_secs(10),
+                Arc::new({
+                    let start = start.clone();
+                    move |s: _| {
+                        Box::pin({
+                            let start = start.clone();
+                            async move {
+                                s.change(format!("ts_{}", start.elapsed().as_secs()).to_owned())
+                                    .await
+                            }
+                        })
+                    }
+                }),
+            )
+            .unwrap();
+
+        // immediate action
+        ctx.actions.select_action().await.unwrap().1(&mut ctx)
+            .await
+            .unwrap();
+
+        // at this point in time there's 7seconds until action_0 and 10seconds until action_1
+
+        ctx.actions.select_action().await.unwrap().1(&mut ctx)
+            .await
+            .unwrap();
+        ctx.actions.select_action().await.unwrap().1(&mut ctx)
+            .await
+            .unwrap();
```

Comment:
Why special case for ts_10 and ts_13? You can just add them to the vector and loop 6 times instead of 4